acme/dns_challenge.go

```diff
@@ -19,37 +20,49 @@ var preCheckDNS preCheckDNSFunc = checkDNS
 
 var preCheckDNSFallbackCount = 5
 
-// DNSProvider represents a service for managing DNS records.
-type DNSProvider interface {
-	CreateTXTRecord(fqdn, value string, ttl int) error
-	RemoveTXTRecord(fqdn, value string, ttl int) error
+// DNSRecord returns the DNS record which will fulfill a DNS challenge
+func DNSRecord(domain, keyAuth string) (fqdn string, value string, ttl int) {
```

Comment:
I'm not sure i follow. Can you be more specific which interface should have comments/docstrings mentioning this function? Or are you just saying this should be more specific that it's for the `dns-01` challenge?

(There is a similar helper functions for generating a cert for `tls-sni-01`)